### PR TITLE
Multi-keystroke sequence now only display the first letter

### LINKED ISSUE #18: Only show the first character in a multi-character sequence by default
When there are more options than the default number of tokens (26 of them), vim-snipe uses a multi-keystroke sequence. Right now it shows the full sequence, which has the issue of shifting the text around as it converts a single character into multiple ones. This makes it hard to use the plugin as the positioning of the character is the one thing the user has to remember which position to pick.

For example if the tokens are "asd" (for simplicity), and the line is ` 12 34 56 78`, then doing (snipe-w) will show ` a2 s4 da6 ds8`, which makes it hard to parse. Ideally it would be ` a2 s4 d6 d8` instead (then when you press 'd', it will show ` 12 34 a6 s8` for further filtering down).

Also, see #16 for examples.

I have a pull request incoming, but wanted to get some thoughts on this first. It will change the default to the new behavior but provide backward compatability. I could switch the default around as well.

### PR DESCRIPTION
**Pull request by ychin**
This avoids shifting column positions of the line in multi-keystroke sequences. Provide option to display the entire sequence like before.

Personally I found the new defaults to make a lot more sense. If you prefer me to use the old defaults instead, I can definitely change the new behavior to be only turned on by the option.

Resolves #18

### CODE REVIEW

File: autoload/snipe/core.vim [hunk 1]
```diff
@@ -168,6 +171,9 @@ function! s:GetJumpCol(jump_tree) " {{{
   for [jump_seq, jump_col] in jump_items
     " this loop builds the highlighted line, adding highlights from left to right;
     " previous multi-token jump sequences are accounted for by tracking col_offset.
+    if !g:snipe_display_full_jump_seq
+        let jump_seq = jump_seq[0]
+    endif
     let len_jump_seq = strlen(jump_seq)
```
Comment (globalchubby) [inline autoload/snipe/core.vim]: Can we use a new variable instead of reassigning `jump_seq`?

***

We should also look for simplifications in the subsequent code:

* `len_jump_seq` will always be `1`; actually, do we even need it now?
* do we still need to track `col_offset`?
* can we get rid of `hl_cols` and just `call add(hl_ids, matchaddpos(g:snipe_hl1_group, [[lnum, jump_seq]]))`?

File: doc/snipe.txt [hunk 2]
```diff
@@ -75,6 +83,29 @@ terminal mode respectively):
     let g:snipe_highlight_gui_color      = '#dfffaf'
     let g:snipe_highlight_cterm256_color = 'blue'
     let g:snipe_highlight_cterm_color    = 'red'
+
```
Comment (globalchubby) [inline doc/snipe.txt]: As I mentioned in https://github.com/yangmillstheory/vim-snipe/issues/18#issuecomment-349472279, do we need to allow users to opt in to this behavior? Your suggested approach is simpler and more intuitive. Having it as the default would give us a smaller API and less documentation to maintain. 

Otherwise, we should add a similar note in the README.

Comment (ychin) [reply doc/snipe.txt]: Yeah that makes sense. I just didn't know if you really wanted that old "display whole sequence" feature. I will clean the code up to not have the option

**Comment by ychin** [comment]
Just pushed new changes to make the new default the only way. Removed the option to opt-in to old behavior, so the plugin will now only show one token at a time.

**Comment by globalchubby** [comment]
I've tested your branch with `let g:snipe_jump_tokens = 'asd'`. Works great, and the simplification is a win!

**Review by globalchubby — APPROVED** [review]

**Comment by globalchubby** [comment]
This is actually new, in the sense that I haven't seen the other related plugins like easymotion or tmux-fingers use this approach.
